KsApi/models/lenses/Project.StatsLenses.swift

```diff
@@ -4,44 +4,67 @@ extension Project.Stats {
     )
 
     public static let goal = Lens<Project.Stats, Int>(
       view: { $0.goal },
-      set: { .init(backersCount: $1.backersCount, commentsCount: $1.commentsCount, goal: $0,
-        pledged: $1.pledged, staticUsdRate: $1.staticUsdRate, updatesCount: $1.updatesCount) }
+      set: { .init(backersCount: $1.backersCount, commentsCount: $1.commentsCount,
+                   currentCurrency:$1.currentCurrency, currentCurrencyRate: $1.currentCurrencyRate, goal: $0,
+                   pledged: $1.pledged, staticUsdRate: $1.staticUsdRate, updatesCount: $1.updatesCount) }
     )
 
     public static let pledged = Lens<Project.Stats, Int>(
       view: { $0.pledged },
-      set: { .init(backersCount: $1.backersCount, commentsCount: $1.commentsCount, goal: $1.goal,
-        pledged: $0, staticUsdRate: $1.staticUsdRate, updatesCount: $1.updatesCount) }
+      set: { .init(backersCount: $1.backersCount, commentsCount: $1.commentsCount,
+                   currentCurrency: $1.currentCurrency, currentCurrencyRate: $1.currentCurrencyRate,
+                   goal: $1.goal, pledged: $0, staticUsdRate: $1.staticUsdRate,
+                   updatesCount: $1.updatesCount) }
     )
 
     public static let staticUsdRate = Lens<Project.Stats, Float>(
       view: { $0.staticUsdRate },
-      set: { .init(backersCount: $1.backersCount, commentsCount: $1.commentsCount, goal: $1.goal,
-        pledged: $1.pledged, staticUsdRate: $0, updatesCount: $1.updatesCount) }
+      set: { .init(backersCount: $1.backersCount, commentsCount: $1.commentsCount,
+                   currentCurrency: $1.currentCurrency, currentCurrencyRate: $1.currentCurrencyRate,
+                   goal: $1.goal, pledged: $1.pledged, staticUsdRate: $0, updatesCount: $1.updatesCount) }
     )
 
     public static let updatesCount = Lens<Project.Stats, Int?>(
       view: { $0.updatesCount },
-      set: { .init(backersCount: $1.backersCount, commentsCount: $1.commentsCount, goal: $1.goal,
-        pledged: $1.pledged, staticUsdRate: $1.staticUsdRate, updatesCount: $0) }
+      set: { .init(backersCount: $1.backersCount, commentsCount: $1.commentsCount,
+                   currentCurrency: $1.currentCurrency, currentCurrencyRate: $1.currentCurrencyRate,
+                   goal: $1.goal, pledged: $1.pledged, staticUsdRate: $1.staticUsdRate, updatesCount: $0) }
     )
 
     public static let fundingProgress = Lens<Project.Stats, Float>(
       view: { $0.fundingProgress },
-      set: { .init(backersCount: $1.backersCount, commentsCount: $1.commentsCount, goal: $1.goal,
-        pledged: Int($0 * Float($1.goal)), staticUsdRate: $1.staticUsdRate, updatesCount: $1.updatesCount) }
+      set: { .init(backersCount: $1.backersCount, commentsCount: $1.commentsCount,
+                   currentCurrency: $1.currentCurrency, currentCurrencyRate: $1.currentCurrencyRate,
+                   goal: $1.goal, pledged: Int($0 * Float($1.goal)), staticUsdRate: $1.staticUsdRate,
```

Comment:
Feels a little strange that this bit `Int($0 * Float($1.goal))` is in the lens, should we move it to the model? 🤔

Reply:
This is a lens we already had on `fundingProgress`, which is a computed property on our model. Gotta have the inverse logic here to update the type accordingly.